File: javascript/rotational-cipher/rotational-cipher.js
const alpha = 'abcdefghijklmnopqrstuvwxyz';

function transform(key, char) {
    let idx = alpha.indexOf(char.toLowerCase());
    if (idx < 0) {
        return char;
    }
    let sub = cycle(idx + key);
    return (alpha.includes(char)) ? sub : sub.toUpperCase();
}

function cycle(distance) {
    return alpha[distance % alpha.length];
}

class RotationalCipher {

    rotate(input, key) {
        return [...input]
            .map(transform.bind(null, key))
            .join('');
    }

}

module.exports = RotationalCipher;

File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Linq;

public static class RotationalCipher
{
    private const string Lower = "abcdefghijklmnopqrstuvwxyz";
    private const string Upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public static string Rotate(string text, int key) => 
        string.Join("", text.Select(c => Rotate(c, key)));

    private static char Rotate(char c, int key) {
        if (!Char.IsLetter(c)) {
            return c;
        }
        string alphabet = Char.IsLower(c) ? Lower : Upper;
        int idx = alphabet.IndexOf(c);
        return alphabet[(idx + key) % alphabet.Length];
    }
}